tests/functional/targets/test_performance.py

```diff
@@ -0,0 +1,131 @@
+"""Performance and Consistence adding and removing targets feature tests."""
+import random
+import time
+from datetime import datetime
+
+import dateutil.parser
+import names_generator
+from pyblake2 import blake2b
+from pytest_bdd import given, scenario, then, when
+from pytest_bdd.parsers import parse
+
+
+@scenario(
+    "../../features/targets/performance.feature",
+    "Multiple requests with multiple targets and timeout threshold",
+)
+def test_api_requester_multiple_request_and_targets():
+    """Multiple requests with multiple targets and timeout threshold"""
+
+
+@given("admin provided the token to the API requester", target_fixture="token")
+def token(access_token):
+    return access_token
+
+
+@given(
+    "the API requester has generated the 'headers' with the token",
+    target_fixture="headers",
+)
+def generate_headers(token):
+    header_token = {"Authorization": f"Bearer {token}"}
+
+    return header_token
+
+
+@given(
+    parse(
+        "the API requester sends {num_requests} requests with {num_targets} "
+        "targets to RSTUF"
+    ),
+    target_fixture="multiple_requests",
+)
+def send_requests_with_targets(
+    headers, num_requests, num_targets, http_request
+):
+    def multiple_targets(number_of_targets):
+        targets = []
+        for count in range(0, int(number_of_targets)):
+            filename = f"test/{names_generator.generate_name()}-{count}.tar.gz"
+            target = {
+                "info": {
+                    "length": random.randint(999, 99999),
+                    "hashes": {
+                        "blake2b-256": blake2b(
+                            filename.encode(), digest_size=32
+                        ).hexdigest()
+                    },
+                },
+                "path": filename,
+            }
```

Comment:
Not sure why you spend CPU time on generating unique filename, length or hashes.
We know that neither the length or the hashes will correspond to the true length or hashes and as such you can just hard code valid values as I did in https://github.com/vmware/repository-service-tuf/pull/183. 

For filenames you can just use `f"test/file_{count}.tar.gz"` or something similar.

Reply:
In my first implementation, I made the approach you suggested
.
We don't use too much CPU time, and this time to do it. You can see the step time in the report generated by this FT.

About the random length, we could have a static one. But the cost here needs to be more relevant, IMO.

It is good to have unique names because we send multiple requests multiple times so that we will ship the same target multiple times. For testing the consistency of RSTUF, it is good to have uniqueness.
Keep in mind that this is just one Scenario. We want to implement more such as sending multiple requests, asynchronous, etc. It is good we force our DB SQL and the metadata to have multiple values to see how it affects overall performance.

Reply:
Okay, then let's focus on what is important to be unique - the file names.
The code will be simpler if we use some dummy values for the other attributes.
We would remove some imports like `random` and `blacke2b` as well.